src/main/java/org/testng/internal/TestResult.java

```diff
@@ -101,26 +102,34 @@ private void init(ITestNGMethod method, ITestContext ctx, Throwable t, long star
     // toString() if it's been overridden.
     if (instance == null) {
       m_name = m_method.getMethodName();
+      return;
+    }
+    if (instance instanceof ITest) {
+      m_name = ((ITest) instance).getTestName();
+      if (m_name == null) {
+        String msg = String.format(
+            "Warning: [%s] implementation on class [%s] returned null. Defaulting to method name",
+            ITest.class.getName(), instance.getClass().getName());
+        if (TestRunner.getVerbose() > 1) {
+          System.err.println(msg);

```

Comment:
You can use the logger class.

Reply:
Using `Utils.log()` was my first choice. But I couldn't get the messages to show up on the console. That was why I chose the `syserr` route.